fs/sync/sync.go

```diff
@@ -1079,6 +1086,16 @@ func (s *syncCopyMove) DstOnly(dst fs.DirEntry) (recurse bool) {
 	return false
 }
 
+// keeps track of dirs with changed contents, to avoid setting modtimes on dirs that haven't changed
```

Comment:
I would make an alternate version of this `markDirModifiedObject` which just takes an `fs.Object` and does the path manipulation and calls `markDirModified`.

Note that `path.Dir` doesn't understand the convention rclone uses (`""` for root directory) so some care is needed using it.

You'll probably need something like

```go
dir := path.Dir(o.Remote())
if dir == "." {
    dir = ""
}
```

Reply:
Got it. Thoughts on making this a new helper function in `fspath`? I frequently find myself reinventing the wheel on path manipulation stuff -- sometimes incorrectly! `fspath` usually has something pretty close but not exactly what I need.

Reply:
I took a stab at this -- see what you think.

Reply:
I think this is great, but `fspath.Dir` is not correct.

Perhaps put in what I suggested then in a separate PR we can find all the `path.Dir` and `path.Base` and work out whether we should be fixing them up with a new function?

Reply:
Ok, made that change.